modules/phase_field/src/userobjects/PolycrystalVoronoi.C

```diff
@@ -28,6 +28,14 @@ PolycrystalVoronoi::validParams()
   params.addParam<unsigned int>("rand_seed", 0, "The random seed");
   params.addParam<bool>(
       "columnar_3D", false, "3D microstructure will be columnar in the z-direction?");
+  params.addParam<bool>(
+      "use_kdtree", true, "Whether or not to use a KD tree to speedup grain search");
```

Comment:
Does using the KD tree search cause any changes to how order parameters are assigned to grains? If so, we ought to think about whether we want to make this the default to true... it look like some existing tests need to be re-golded, is that because of differences in OP to grain assignment? I'm just worried users might be trying to reproduce previous results and run into troubles doing it because of the change. I guess they can always set this to false to reproduce the previous results.

Reply:
Yes, we may set use_kdtree to false by default. If there is a unique solution, kdtree will give the same result as the naive method. When there are multiple nearest points,  kdtree may provide a different solution from the naive approach. All the results are equally bad or good. But users can complain about that for sure.

So we might not use kdtree by default. But users will have to stay with the slow code ...

Producibility might be more important than efficiency ...

Reply:
Yeah I feel like if we can just mention whichever way we want to go as the default in the documentation that KDtree is an option that can be used to speed up IC generation for larger systems, it should be clear to users. If we use KDtree by default, as long as the documentation makes it clear that to reproduce previous results the user may need to set it to false, that would be totally fine.

Reply:
Yeah, I think we should set `use_kdtree = false` as the default option. So, we don't affect reproducibility. Additionally, I would recommend that the users should check what `grain_patch_size` would be ideal for their problem, before using this option. I will add this to the documentation.